modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModalDeleteObjectDefinition.tsx

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayModalProvider, useModal} from '@clayui/modal';
+import {Observer} from '@clayui/modal/lib/types';
+import {createResourceURL, fetch} from 'frontend-js-web';
+import React, {useEffect, useState} from 'react';
+
+import DangerModal from './DangerModal';
+import WarningModal from './WarningModal';
+
+function ModalDeleteObjectDefinition({
+	objectDefinition,
+	observer,
+	onClose,
+	onDelete,
+}: IProps) {
+	const {
+		hasObjectRelationship,
+		id,
+		name,
+		objectEntriesCount,
+		status: {code},
+	} = objectDefinition;
+
+	return hasObjectRelationship ? (
+		<WarningModal
+			observer={observer}
+			onClose={onClose}
+			title={Liferay.Language.get('deletion-not-allowed')}
+		>
+			<div>
+				{Liferay.Language.get(
+					'this-object-cannot-be-deleted-because-it-is-related-to-others'
+				)}
+			</div>
+
+			<div>
+				{Liferay.Language.get(
+					'to-delete-this-object-you-need-first-delete-its-relationships'
+				)}
+			</div>
+
+			<div>
+				{Liferay.Language.get('go-to-object-details-relationships')}
+			</div>
+		</WarningModal>
+	) : code === 0 ? (
```

Comment:
A ternary into another ternary may affect the readability of the code. for those cases you better use regular `if` clause:
```suggestion
	if (hasObjectRelationship) {
		return (
			<WarningModal
				observer={observer}
				onClose={onClose}
				title={Liferay.Language.get('deletion-not-allowed')}
			>
				<div>
					{Liferay.Language.get(
						'this-object-cannot-be-deleted-because-it-is-related-to-others'
					)}
				</div>

				<div>
					{Liferay.Language.get(
						'to-delete-this-object-you-need-first-delete-its-relationships'
					)}
				</div>

				<div>
					{Liferay.Language.get('go-to-object-details-relationships')}
				</div>
			</WarningModal>
		) ;
	} 
	
	return code === 0 ? (
```